Backend/AMQPBackendDispatcher.php

```diff
@@ -55,25 +57,48 @@ public function __construct(array $settings, array $queues, $defaultQueue, array
     }
 
     /**
-     * @return AMQPChannel
+     * @return AmqpContext
      */
-    public function getChannel()
+    public function getContext()
     {
-        if (!$this->channel) {
-            $this->connection = new AMQPConnection(
-                $this->settings['host'],
-                $this->settings['port'],
-                $this->settings['user'],
-                $this->settings['pass'],
-                $this->settings['vhost']
-            );
+        if (!$this->context) {
+            if (false == array_key_exists('factory_class', $this->settings)) {
+                throw new \LogicException('The factory_class option is missing though it is required.');
+            }
+
+            $factoryClass = $this->settings['factory_class'];
+            if (false == class_exists($factoryClass) || false == (new \ReflectionClass($factoryClass))->implementsInterface(AmqpConnectionFactory::class)) {
+                throw new \LogicException(sprintf('The factory_class option has to be valid class that implements "%s"', AmqpConnectionFactory::class));
+            }
+
+            /** @var AmqpConnectionFactory $factory */
+            $factory = new $factoryClass([
+                'host' => $this->settings['host'],
+                'port' => $this->settings['port'],
+                'user' => $this->settings['user'],
+                'pass' => $this->settings['pass'],
+                'vhost' => $this->settings['vhost'],
+            ]);
+
+            if ($factory instanceof DelayStrategyAware) {
+                $factory->setDelayStrategy(new RabbitMqDlxDelayStrategy());
+            }
 
-            $this->channel = $this->connection->channel();
+            $this->context = $this->connectionFactory->createContext();
 
             register_shutdown_function([$this, 'shutdown']);
         }
 
-        return $this->channel;
+        return $this->context;
+    }
+
+
+    /**
+     * @deprecated
+     */
+    public function getChannel()
+    {
+        throw new \LogicException('Not used any more');
```

Comment:
That looks like a BC-break. Should it call `getContext()` instead?

Reply:
Forgot to remove it.

Reply:
There are some BC breaks in case of a user extends classes. The API is the same though.